lib/notification-element.coffee

```diff
@@ -233,16 +236,25 @@ class NotificationElement
     else
       Promise.resolve()
 
+  makeDismissable: ->
+    unless @model.isDismissable()
+      clearTimeout(@autohideTimeout)
+      @model.options.dismissable = true
+      @element.classList.add('has-close')
+
   removeNotification: ->
```

Comment:
I notice you're calling this method in more places now with this PR.  Can you describe to me why that's necessary?

Reply:
because a non-dismissable notification won't emit an `onDidDismiss` event when `.dismiss()` is called on it.

And there are multiple times we would want to remove the notification.
  1. if `.dismiss()` is called on the notification by another package
  2. if the `x` is clicked in the corner
  3. if the 'Close All' button is clicked
  4. if the autohide timer is up